src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -73,6 +391,42 @@ def __init__(self, sc, *args):
             self.sqlCtx = SQLContext(sc)
         except Py4JError:
             traceback.print_exc()
+            
+    def start(self):
+        try:
+            prevGlobalMLContext = MLContext.globalMLContext
+        except AttributeError:
+            pass
+        self.numTabs = 0
+        self.sqlCtx = SQLContext(self.sc)
+        MLContext.globalMLContext = self
+        self.reset()
+        self.pydml = []
+        self.matrixOutput = []
+        
+    
+    def eval(self, outputs=None):
```

Comment:
Can you update `outputs=None` to `*outputs` to allow for a variable number of outputs, which could be empty, and will always be suppled to the body of the function as a list?  This would mean that this would be called as `ml.eval(m1, m2)`, for example, or as `ml.eval()`.  Then you can do the following:

```
for output in outputs:
  output.registerAsOutput()

for o in self.matrixOutput():
  self.registerOutput(o.getID())
  addPyDMLStmt('save', args=[o.getID(), ', \" \"'])
```